lighthouse/classes/event_properties/definitions.py

```diff
@@ -257,3 +280,224 @@ def value(self):
 
     def add_to_warehouse_message(self, message):
         message.add_metadata("source_plate_barcode", self.value)
+
+
+class CherrytrackWellsFromDestination(EventPropertyAbstract, ServiceCherrytrackMixin):
+    def __init__(self, barcode_property: PlateBarcode):
+        self.reset()
+        self.barcode_property = barcode_property
+
+    def validate(self):
+        return self.barcode_property.validate() and (len(self._errors) == 0)
+
+    def _validate_destination_coordinate_not_duplicated(self, wells):
+        coordinates = [well["destination_coordinate"] for well in wells]
+        duplicates = set([coor for coor in coordinates if coordinates.count(coor) > 1])
+        if len(duplicates) > 0:
+            raise RetrievalError(f"Some coordinates have clashing samples/controls: { duplicates }")
+
+    @property
+    def errors(self) -> List[str]:
+        self.validate()
+        return self._errors + self.barcode_property.errors
+
+    @cached_property
+    def value(self):
+        with self.retrieval_scope():
+            val = self.get_wells_from_destination_plate(self.barcode_property.value)
+            self._validate_destination_coordinate_not_duplicated(val)
+            return val
+
+    def add_to_warehouse_message(self, message):
+        return None
```

Comment:
I think this should be `pass` rather: https://docs.python.org/3/reference/simple_stmts.html#the-pass-statement

Reply:
ok done